scheduler/test/cook/test/mesos/scheduler.clj

```diff
@@ -2175,5 +2175,36 @@
           (is (= :executor/cook (:instance/executor instance-ent)))
           (is (nil? (:instance/sandbox-directory instance-ent))))))))
 
+
+(deftest test-monitor-tx-report-queue
+  (let [conn (restore-fresh-database! "datomic:mem://test-monitor-tx-report-queue")
+        job-id-1 (create-dummy-job conn)
+        instance-id-1 (create-dummy-instance conn job-id-1)
+        job-id-2 (create-dummy-job conn)
+        instance-id-2 (create-dummy-instance conn job-id-2)
+        killed-tasks (atom [])
+        mock-driver (reify msched/SchedulerDriver
+                      (kill-task! [_ task-id]
+                        (swap! killed-tasks #(conj % task-id))))
+        driver-ref (ref mock-driver)]
+    (cook.mesos/kill-job conn [(:job/uuid (d/entity (d/db conn) job-id-1))])
+    (let [job-ent (d/entity (d/db conn) job-id-1)
+          instance-ent (d/entity (d/db conn) instance-id-1)]
+      (is (= :job.state/completed (:job/state job-ent)))
+      (is (= :instance.status/unknown (:instance/status instance-ent))))
+    (let [[report-mult close-fn] (cook.datomic/create-tx-report-mult conn)
+          transaction-chan (async/chan (async/sliding-buffer 4096))
+          _ (async/tap report-mult transaction-chan)
+          kill-fn (cook.mesos.scheduler/monitor-tx-report-queue transaction-chan conn driver-ref)]
+      (cook.mesos/kill-job conn [(:job/uuid (d/entity (d/db conn) job-id-2))])
+      ;; TODO - find a better way to wait for this
+      (Thread/sleep 100)
```

Comment:
Can we use the "wait until" pattern to wait until `killed-tasks` has what we expect?

Reply:
IMHO, it's overkill here. The 100 ms should be plenty, I just was trying to think of something to programmatically make sure we have gone through the entire transaction queue

Reply:
What's the downside to using "wait until"?

Reply:
It just seems like every sleep we introduce is asking for flakiness at some point down the road, maybe on Travis

Reply:
Agreed, added